controller/dataplane/controller_reconciler_utils.go

```diff
@@ -78,6 +84,169 @@ func (r *Reconciler) ensureDataPlaneAddressesStatus(
+			return nil, reconciliationResult, err
+		}
+		configMapsToRetain[cp.ConfigMapNN] = struct{}{}
+		cps = append(cps, cp)
+	}
+	for _, cm := range configMapsOwned {
+		if _, retain := configMapsToRetain[client.ObjectKeyFromObject(&cm)]; !retain {
+			if err := c.Delete(ctx, &cm); err != nil {
+				if k8serrors.IsNotFound(err) {
+					continue
+				}
+				return nil, ctrl.Result{}, err
+			}
+		}
+	}
+
+	return withCustomPlugins(cps), ctrl.Result{}, nil
+}
+
+func findCustomPluginConfigMapsOwnedByDataPlane(
+	ctx context.Context, c client.Client, dataplane *operatorv1beta1.DataPlane,
+) ([]corev1.ConfigMap, error) {
+	cms, err := k8sutils.ListConfigMapsForOwner(ctx, c, dataplane.GetUID())
+	if err != nil {
+		return nil, err
+	}
+	return lo.Filter(cms, func(cm corev1.ConfigMap, _ int) bool {
+		_, isForKPI := cm.Annotations[consts.AnnotationKongPluginInstallationMappedKongPluginInstallation]
+		return isForKPI
+	}), nil
+}
+
+func populateDedicatedConfigMapForKongPluginInstallation(
+	ctx context.Context,
+	logger logr.Logger,
+	c client.Client,
+	cms []corev1.ConfigMap,
+	kpiNN types.NamespacedName,
+	dataplane *operatorv1beta1.DataPlane,
+) (customPlugin, ctrl.Result, error) {
+	kpi, ready, err := verifyKPIReadinessForDataPlane(ctx, logger, c, dataplane, kpiNN)
+	if err != nil {
+		return customPlugin{}, ctrl.Result{}, err
+	}
+	if !ready {
+		return customPlugin{}, ctrl.Result{Requeue: true}, nil
+	}
+
+	log.Trace(logger, "Fetch underlying ConfigMap for KongPluginInstallation", kpi)
+	var underlyingCM corev1.ConfigMap
+	backingCMNN := types.NamespacedName{
+		Namespace: kpi.Namespace,
+		Name:      kpi.Status.UnderlyingConfigMapName,
+	}
+	if err := c.Get(ctx, backingCMNN, &underlyingCM); err != nil {
+		return customPlugin{}, ctrl.Result{}, fmt.Errorf("could not fetch underlying ConfigMap to clone %s: %w", backingCMNN, err)
+	}
+
+	log.Trace(logger, "Find ConfigMap mapped to KongPluginInstallation", kpi)
+	mappedConfigMapForKPI := lo.Filter(cms, func(cm corev1.ConfigMap, _ int) bool {
+		kpiNN := cm.Annotations[consts.AnnotationKongPluginInstallationMappedKongPluginInstallation]
+		return kpiNN == client.ObjectKeyFromObject(&kpi).String()
+	})
+	var cm corev1.ConfigMap
+	switch len(mappedConfigMapForKPI) {
+	case 0:
+		log.Trace(logger, "Create new ConfigMap for KongPluginInstallation", kpi)
+		cm.GenerateName = dataplane.Name + "-"
+		cm.Namespace = dataplane.Namespace
+		k8sutils.SetOwnerForObject(&cm, dataplane)
+		k8sresources.LabelObjectAsDataPlaneManaged(&cm)
+		k8sresources.AnnotateConfigMapWithKongPluginInstallation(&cm, kpi)
+		cm.Data = underlyingCM.Data
+		if err := c.Create(ctx, &cm); err != nil {
+			return customPlugin{}, ctrl.Result{}, fmt.Errorf("could not create new mapped: %w", err)
+		}
+	case 1:
+		log.Trace(logger, "Update existing ConfigMap for KongPluginInstallation", kpi)
+		cm = mappedConfigMapForKPI[0]
+		cm.Data = underlyingCM.Data
+		if err := c.Update(ctx, &cm); err != nil {
```

Comment:
If `k8serrors.IsConflict(err)` I'd suggest to not return an error but a `ctrl.Result{Requeue: true}` to not bother users with an error that most likely stems from the operator synchronization (because of using a cached client). WDYT?

Reply:
Very good suggestion, btw. we should probably go through our codebase to find some spots like that, because even when you look at logs of our automated tests we can see some transitive errors that may not should bother users